README.md

```diff
@@ -195,16 +194,31 @@ by replacing
 
 by
 
+### Modifying the NameId sent in the SAML token
+
+You can change the way the NameId is built by changing the [saml-nameid.xml](idp/shibboleth-idp/conf/saml-nameid.xml).
+There is a bean declaration of `SAML2NameIDGenerators` and the NameId construction is specified
+at line 37 with `p:attributeSourceIds="#{ {'mail'} }"`.
+
+To have an empty NameId, change it to:
+`p:attributeSourceIds="#{ '' }"`
+
+To have a NameId that is not the email but the uid, change it to:
+`p:attributeSourceIds="#{ {'uid'} }"`
+
+I did not test, but I believe you could build whatever NameId you need based on users properties
+from [users.ldif](ldap/users.ldif).
+
+You will need to rebuild the project to take this changes in account.
 
 ```xml
 <resolver:AttributeEncoder xsi:type="enc:SAML2String" name="surname" friendlyName="sn" encodeType="false" />
 ```
 After doing so, you will need to rebuild the project:
 
 ```bash
-docker-compose rm
-docker-compose build
-docker-compose up
+docker-compose down
```

Comment:
I think you really need the `rm` (think about your dev-proxy issue)

Reply:
The `docker-compose down` commands stops the containers, removes them, then trashes the networks.
actually:
- `docker-compose stop` stops containers
- `docker-compose rm` removes the containers but not the networks.

Reply:
All my tests worked with just `docker-compose down` and `docker-compose up --build` as it only rebuilt the needed layers with file modifications.